postprocessing/science/compute_diff_seissol_data.py

```diff
@@ -0,0 +1,201 @@
+    """
+    ind = fuzzysort(geom.T, list(range(0, geom.shape[0])), tol=1e-4)
+    # generate inverse look-up table
+    dic = {i: index for i, index in enumerate(ind)}
+    ind_inv = np.zeros_like(ind)
+    for k, v in dic.items():
+        ind_inv[v] = k
+    return ind, ind_inv
+
+
+def read_geom_connect(sx):
+    return sx.ReadGeometry(), sx.ReadConnect()
+
+
+def return_sorted_geom_connect(sx):
+    """sort geom array and reindex connect array to match the new geom array"""
+    geom, connect = read_geom_connect(sx)
+    nv = geom.shape[0]
+
+    try:
+        import pymesh
+
+        geom, connect, inf = pymesh.remove_duplicated_vertices_raw(
+            geom, connect, tol=1e-4
+        )
+        print(f"removed {inf['num_vertex_merged']} duplicates out of {nv}")
+    except ModuleNotFoundError:
+        print("pymesh not found, trying trimesh...")
+        import trimesh
+
+        trimesh.tol.merge = 1e-4
+        mesh = trimesh.Trimesh(geom, connect)
+        mesh.merge_vertices()
+        geom = mesh.vertices
+        connect = mesh.faces
+        print(f"removed {nv-geom.shape[0]} duplicates out of {nv}")
+
+    ind, ind_inv = lookup_sorted_geom(geom)
+    geom = geom[ind, :]
+    connect = np.array([ind_inv[x] for x in connect.flatten()]).reshape(connect.shape)
+    # sort along line (then we can use multidim_intersect)
+    connect = np.sort(connect, axis=1)
+    return geom, connect
+
+
+def multidim_intersect(arr1, arr2):
+    """find indexes of same triangles in 2 connect arrays
+    (associated with the same geom array)
+    generate 1D arrays of tuples and use numpy function
+    https://stackoverflow.com/questions/9269681/intersection-of-2d-numpy-ndarrays
+    """
+    arr1_view = arr1.view([("", arr1.dtype)] * arr1.shape[1])
+    arr2_view = arr2.view([("", arr2.dtype)] * arr2.shape[1])
+    intersected, ind1, ind2 = np.intersect1d(arr1_view, arr2_view, return_indices=True)
+    ni, n1, n2 = intersected.shape[0], arr1.shape[0], arr2.shape[0]
+    print(
+        f"{ni} faces in common, n faces connect 1:{n1}, 2:{n2} (diff: {n1-ni}, {n2-ni})"
+    )
+    return ind1, ind2
+
+
+def same_geometry(sx1, sx2):
+    geom1 = sx1.ReadGeometry()
+    geom2 = sx2.ReadGeometry()
+    if geom1.shape[0] != geom2.shape[0]:
+        return False
+    else:
+        return np.all(np.isclose(geom1, geom2, rtol=1e-3, atol=1e-4))
+
+
+parser = argparse.ArgumentParser(
+    description="make difference between 2 (paraview) output files: f2-f1. \
+The output must be from the same mesh, but the partionning may differ."
+)
+parser.add_argument("xdmf_filename1", help="filename1")
+parser.add_argument("xdmf_filename2", help="filename2")
+parser.add_argument(
+    "--idt",
+    nargs="+",
+    required=True,
+    help="list of time step to differenciate (1st = 0); -1 = all",
+    type=int,
+)
+parser.add_argument(
+    "--Data",
+    nargs="+",
+    required=True,
+    metavar=("variable"),
+    help="Data to differenciate (example SRs)",
```

Comment:
Can you also add an "all" option here?

Reply:
I added your commit from the DR branch, and improved it a little bit.